ArdeleanuStefan_Task3/server.cpp

```diff
@@ -0,0 +1,135 @@
+#include <boost/interprocess/ipc/message_queue.hpp>
+#include <boost/interprocess/managed_shared_memory.hpp>
+#include <boost/interprocess/containers/string.hpp>
+#include <boost/interprocess/containers/vector.hpp>
+#include <boost/interprocess/allocators/allocator.hpp>
+#include <iostream>
+#include <boost/asio.hpp>
+#include <DynamicStringArray.hpp>
+#include "common.h"
+
+using namespace boost::interprocess;
+
+namespace Shared
+{
+   using Segment = managed_shared_memory;
+   using Manager = Segment::segment_manager;
+
+   using CharAllocator = allocator<char, Manager>;
+   using String = basic_string<char, std::char_traits<char>, CharAllocator>;
+
+   using StringAllocator = allocator<String, Manager>;
+   using Vector = vector<String, StringAllocator>;
+}
+
+class Server
+{
+private:
+   DynamicStringArray m_string_array;
+   std::unique_ptr<message_queue> m_mq;
+   std::unique_ptr<managed_shared_memory> m_managed_smh;
+
+public:
+   Server()
+   {
+      shared_memory_object::remove(SHARED_MEMORY_NAME.c_str());
+      message_queue::remove(MESSAGE_QUEUE_NAME.c_str());
+
+      m_mq = std::make_unique<message_queue>(create_only, MESSAGE_QUEUE_NAME.c_str(), MAX_MESSAGE_NUMBER, MAX_MESSAGE_SIZE);
+      m_managed_smh = std::make_unique<managed_shared_memory>(create_only, SHARED_MEMORY_NAME.c_str(), SHARED_MEMORY_SIZE);
+   }
+
+   void start()
+   {
```

Comment:
In this method there is a lot of common code, maybe you should consider to use inheritance and write only one external function.